pytorch_lightning/metrics/classification/precision_recall.py

```diff
@@ -11,205 +11,336 @@
+
+        threshold:
+            Threshold probability value for transforming probability predictions to binary
+            (0,1) predictions, in the case of binary or multi-label inputs.
+        top_k:
+            Number of highest probability entries for each sample to convert to 1s - relevant
+            only for inputs with probability predictions. If this parameter is set for multi-label
+            inputs, it will take precedence over ``threshold``. For (multi-dim) multi-class inputs,
+            this parameter defaults to 1.
+
+            Should be left unset (``None``) for inputs with label predictions.
+        is_multiclass:
+            Used only in certain special cases, where you want to treat inputs as a different type
+            than what they appear to be. See the parameter's
+            :ref:`documentation section <metrics:Using the is_multiclass parameter>`
+            for a more detailed explanation and examples.
 
-        multilabel: If predictions are from multilabel classification.
         compute_on_step:
-            Forward only calls ``update()`` and return None if this is set to False. default: True
+            Forward only calls ``update()`` and return ``None`` if this is set to ``False``.
         dist_sync_on_step:
             Synchronize metric state across processes at each ``forward()``
-            before returning the value at the step. default: False
+            before returning the value at the step
         process_group:
-            Specify the process group on which synchronization is called. default: None (which selects the entire world)
+            Specify the process group on which synchronization is called.
+            default: ``None`` (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state. When ``None``, DDP
+            will be used to perform the allgather.
 
     Example:
 
-        >>> from pytorch_lightning.metrics import Precision
-        >>> target = torch.tensor([0, 1, 2, 0, 1, 2])
-        >>> preds = torch.tensor([0, 2, 1, 0, 0, 1])
-        >>> precision = Precision(num_classes=3)
+        >>> from pytorch_lightning.metrics.classification import Precision
+        >>> preds  = torch.tensor([2, 0, 2, 1])
+        >>> target = torch.tensor([1, 1, 2, 0])
+        >>> precision = Precision(average='macro', num_classes=3)
         >>> precision(preds, target)
-        tensor(0.3333)
+        tensor(0.1667)
+        >>> precision = Precision(average='micro')
+        >>> precision(preds, target)
+        tensor(0.2500)
 
     """
+
     def __init__(
         self,
-        num_classes: int = 1,
+        average: str = "micro",
+        mdmc_average: Optional[str] = None,
+        zero_division: int = 0,
+        ignore_index: Optional[int] = None,
+        num_classes: Optional[int] = None,
         threshold: float = 0.5,
-        average: str = 'micro',
-        multilabel: bool = False,
```

Comment:
we probably need to still have this arg but deprecate it (it should have no effect)

Reply:
IMO if we deprecate it, it should raise a warning, but work as it did before until we remove it.

Reply:
@justusschock This is not needed - the parameter was just there so that the metric could properly format inputs. This is not necessary anymore, the new formatting function can detect if the inputs are multi-label or not. I will deprecate this parameter as @SkafteNicki suggested